packages/widget/src/utils/index.ts

```diff
@@ -46,3 +50,39 @@ export const capitalize = (s: string): string => {
   const rest = s.slice(1);
   return `${firstLetter}${rest}`;
 };
+
+export const validateSubstrateAddress = (address: string): boolean => {
+  try {
+    encodeAddress(isHex(address) ? hexToU8a(address) : decodeAddress(address));
+    return true;
+  } catch (error) {
+    return false;
+  }
+};
+
+export const validateAddress = (address: string, network: Network): string => {
+  let errorMessage: string = '';
+
+  switch (network) {
+    case Network.SUBSTRATE: {
+      const validPolkadotAddress = validateSubstrateAddress(address);
+
+      if (!validPolkadotAddress) {
+        errorMessage = 'invalid Substrate address';
+      }
+      break;
+    }
+    case Network.EVM: {
+      const isAddress = ethers.utils.isAddress(address);
+      if (!isAddress) {
+        errorMessage = 'invalid Ethereum Address';
+      }
+      break;
+    }
+    default:
+      errorMessage = 'invalid address';
+      break;
+  }
+
+  return errorMessage;
+};
```

Comment:
and lastly, this function would look something like this:
```suggestion
export const validateAddress = (address: string, network: Network): string | null => {
  switch (network) {
    case Network.SUBSTRATE: {
      const validPolkadotAddress = validateSubstrateAddress(address);

      return validPolkadotAddress ? null : 'invalid Substrate address';
    }
    case Network.EVM: {
      const isAddress = ethers.utils.isAddress(address);
      
      return isAddress ? null : 'invalid Ethereum address';
    }
    default:
      return 'unsupported network';
  }
};
```